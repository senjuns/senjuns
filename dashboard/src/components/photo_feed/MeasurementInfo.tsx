import Box from '@material-ui/core/Box';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { FC, HTMLAttributes } from 'react';
import TimeHistoryButton from '../../components/photo_section_view/TimeHistoryButton';

const useStyles = makeStyles(() =>
  createStyles({
    infoBox: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-end',
      textTransform: 'uppercase',
    },
  })
);

interface MeasurementInfoProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * The last time when the system has been captured.
   */
  lastCapturedTime?: Date | null;
}

const MeasurementInfo: FC<MeasurementInfoProps> = ({
  lastCapturedTime,
  ...props
}) => {
  const classes = useStyles();

  return (
    <Box className={classes.infoBox} {...props}>
      {lastCapturedTime ? (
        <TimeHistoryButton
          capturedAt={lastCapturedTime}
          isTrackerOpen={true}
          onClick={() => {}}
        />
      ) : (
        <Typography>No Capture Yet</Typography>
      )}
    </Box>
  );
};

export default MeasurementInfo;
