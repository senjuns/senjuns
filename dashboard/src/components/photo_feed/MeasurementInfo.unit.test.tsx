// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import MeasurementInfo from './MeasurementInfo';

describe('<MeasurementInfo />', () => {
  it('renders correctly when the lastCapturedTime is provided.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <MeasurementInfo lastCapturedTime={new Date(2021, 9, 10, 12, 30, 20)} />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders correctly when the lastCapturedTime is null.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<MeasurementInfo lastCapturedTime={null} />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
