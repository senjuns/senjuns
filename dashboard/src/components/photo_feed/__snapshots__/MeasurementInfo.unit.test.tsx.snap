// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasurementInfo /> renders correctly when the lastCapturedTime is null. 1`] = `
<Styled(MuiBox)
  className="makeStyles-infoBox-1"
>
  <WithStyles(ForwardRef(Typography))>
    No Capture Yet
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`<MeasurementInfo /> renders correctly when the lastCapturedTime is provided. 1`] = `
<Styled(MuiBox)
  className="makeStyles-infoBox-1"
>
  <TimeHistoryButton
    capturedAt={2021-10-10T12:30:20.000Z}
    isTrackerOpen={true}
    onClick={[Function]}
  />
</Styled(MuiBox)>
`;
